src/catchup/ApplyCheckpointWork.h

```diff
@@ -63,10 +63,9 @@ class ApplyLedgerChainWork : public BasicWork
     bool applyHistoryOfSingleLedger();
 
   public:
-    ApplyLedgerChainWork(Application& app, TmpDir const& downloadDir,
-                         LedgerRange range,
-                         LedgerHeaderHistoryEntry& lastApplied);
-    ~ApplyLedgerChainWork() = default;
+    ApplyCheckpointWork(Application& app, TmpDir const& downloadDir,
+                        uint32_t checkpoint, uint32_t boundary = 0);
```

Comment:
I don't like the use of 0 like this in a low level class as it's quite error prone.

I think it's probably better to keep a `LedgerRange` here, and force the caller to make a decision on what the upper range is.

This PR then just needs to enforce that that LedgerRange has the following properties:
1. low is a checkpoint
2. doesn't span multiple checkpoints

Reply:
Agreed that it's safer, updated.